x-pack/plugins/triggers_actions_ui/public/common/index_controls/index.ts

```diff
@@ -8,48 +8,30 @@
 import { uniq } from 'lodash';
 import { HttpSetup } from 'kibana/public';
 import { i18n } from '@kbn/i18n';
-import {
-  loadIndexPatterns,
-  getMatchingIndices,
-  getESIndexFields,
-  getSavedObjectsClient,
-} from '../lib/data_apis';
+import { loadIndexPatterns, getMatchingIndices, getESIndexFields } from '../lib/data_apis';
 
 export interface IOption {
   label: string;
   options: Array<{ value: string; label: string }>;
 }
 
-export const getIndexPatterns = async () => {
-  // TODO: Implement a possibility to retrive index patterns different way to be able to expose this in consumer plugins
-  if (getSavedObjectsClient()) {
-    const indexPatternObjects = await loadIndexPatterns();
-    return indexPatternObjects.map((indexPattern: any) => indexPattern.attributes.title);
-  }
-  return [];
-};
-
-export const getIndexOptions = async (
-  http: HttpSetup,
-  pattern: string,
-  indexPatternsParam: string[]
-) => {
+export const getIndexOptions = async (http: HttpSetup, pattern: string) => {
   const options: IOption[] = [];
 
   if (!pattern) {
     return options;
   }
 
-  const matchingIndices = (await getMatchingIndices({
-    pattern,
-    http,
-  })) as string[];
-  const matchingIndexPatterns = indexPatternsParam.filter((anIndexPattern) => {
-    return anIndexPattern.includes(pattern);
-  }) as string[];
+  const [matchingIndices, matchingIndexPatterns] = await Promise.all([
```

Comment:
This comes down to the UX. What should we do if the http request fails? 
Nothing? toast message? message in dropdown menu?
Therefore, i added  try-catch blocks in the api implementation, they never return an error, if one of them fails they return an empty array, since the options are formatted and a choose option for the typed keyword is shown shouldn;t be a problem.

Reply:
I agree with catching the error and returning an empty array. User can try re-typing if they are looking for a specific index pattern and hopefully the request would succeed next time.

Reply:
We were not sure about that toast message, it may shift the focus from input to popup... and since it fetches the results as the user types, the request would be repeated when a new letter is typed :)

In the example you shared the request is made once on component load.

And they can still type the whole index name and select, so it's not a blocker for the user.

Reply:
Sounds good! I'm ok keeping what you have as-is unless others feel differently 👍